src/realm.js

```diff
@@ -232,7 +232,8 @@ export class Realm {
       metadata?: any
     ) => [Completion | Reference | Value, BabelNode, Array<BabelNodeStatement>],
   };
-  simplifyAbstractValue: AbstractValue => Value;
+  simplifyAndRefineAbstractValue: AbstractValue => Value;
+  simplifyAndRefineAbstractCondition: AbstractValue => Value;
```

Comment:
Just from the name of it, I still don't get what `...Condition` does. Is it path sensitive? Is it meant to only be called for particular kinds of conditions? Conditional expressions, or maybe binary conditions?

Reply:
Refinement is path sensitive by definition.

Reply:
I see. Now I understand :-) Can you persist that as a comment?

Reply:
Sure.